app/files.test.js

```diff
@@ -179,6 +182,43 @@ Files("triggers when a file is deleted", async (context) => {
   else equal(true, true)
 })
 
+Files("hidden share roles trigger return roles", async (context) => {
+  const { perform } = shareRoles.operation
+  /** @type {{id: number}} */
+  const inputData = {
+    id: context.inputData.id
+  }
+  const bundle = {
+    authData: context.authData,
+    inputData
+  }
+  const roles = await tester(perform, bundle)
+  const role = roles[0]
+  if (!role) {
+    unreachable("TODO")
+    return
+  }
+  context.inputData.access = role.id
+  not.equal(role.id, 0)
+})
+
+//TODO: Without an action to invite the user by email, there is no user id to test. The test works with hardcode value
+/*
+Files("share a room with a user", async (context) => {
+  const { perform } = shareRoom.operation
+  const inputData = {
+    id: context.inputData.folderId,
+    user: "",
+    access: context.inputData.access
+  }
+  const bundle = {
+    authData: context.authData,
+    inputData
+  }
+  const members = await tester(perform, bundle)
+})
+*/
```

Comment:
I'm not quite sure why we need the user's email. Could we create a new user in the test case, send them the invite, and then check if this user is included in the members of the room?

Reply:
> I'm not quite sure why we need the user's email. Could we create a new user in the test case, send them the invite, and then check if this user is included in the members of the room?

Yes, but I have already done the user invitation by email (PR https://github.com/ONLYOFFICE/onlyoffice-zapier/pull/37), and in my opinion it would make more sense to use the invited user to add them to the room.